pages/index.js

```diff
@@ -1,15 +1,58 @@
+import Header from "@/components/Header";
+import TransactionForm from "@/components/TransactionForm";
 import TransactionsList from "@/components/TransactionsList";
+import { transactions } from "@/lib/transactions";
+import { useState } from "react";
+import styled from "styled-components";
+import { ulid } from "ulid";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function HomePage() {
+  const [transactionsList, setTransactionsList] = useLocalStorageState(
+    "transactions",
+    { defaultValue: transactions }
+  );
+  const [successMessage, setSuccessMessage] = useState("");
+
+  function handleAddTransaction(data) {
+    setTransactionsList([
+      { ...data, id: ulid(), currency: "EUR" },
+      ...transactionsList,
+    ]);
+    setSuccessMessage("Transaction successfully added!");
+    setTimeout(() => {
+      setSuccessMessage("");
+    }, 3000);
```

Comment:
A timeout is a side effect and must therefore always be in a `useEffect`. 👆🤯😅 You also need a cleanup function for the timeout.

Reply:
Thank you, we have added the useEffect.